### PR TITLE
mkfifo implementation

### LINKED ISSUE #34: Implement mkfifo

### PR DESCRIPTION
**Pull request by FedericoPonzi**
Fixes #34

### CODE REVIEW

**Comment by FedericoPonzi** [comment]
@GrayJack Could you help me figure out what's wrong with the build? :thinking:

**Comment by GrayJack** [comment]
Yeah, I put more targets to test on Linux, MacOS and Fuchsia, but forgot that musl targets won't build at all (plus a issue pop-up due me ignoring 32bit while doing some tools, now fixed)

File: coreutils_core/src/mkfifo.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            ));
+        },
+        _ => {
+            return Err(io::Error::new(
+                io::ErrorKind::Other,
+                format!("could not open {:?}: {}", path, error),
+            ));
+        },
+    }
+}
```
Comment (GrayJack) [inline coreutils_core/src/mkfifo.rs]: As far as I know, `Error::last_os_error()` already gets the error kind with the right `Display` for each Unix errno , so I don't think you need all that code overhead, just return `Error::last_os_error()`. In my experience, it didn't failed me in any moment

Comment (FedericoPonzi) [reply coreutils_core/src/mkfifo.rs]: Awesome, I've updated the code accordingly to your comment :)

File: mkfifo/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+use coreutils_core::mkfifo::mkfifo;
+use std::process;
+
+fn main() {
+    let yaml = load_yaml!("mkfifo.yml");
+    let matches = App::from_yaml(yaml).settings(&[ColoredHelp]).get_matches();
+    let filepath = matches.value_of("NAME").expect("No filename provided.");
+    let mode = matches.value_of("mode").unwrap();
+    let mode: u32 = u32::from_str_radix(mode, 8)
+        .unwrap_or_else(|_| panic!("Mode '{}' is not an octal number.", mode));
```
Comment (GrayJack) [inline mkfifo/src/main.rs]: Panic will make the program exit with code 101, the mkfifo tool returns exit code 1 when mode is not a octal number

Comment (FedericoPonzi) [reply mkfifo/src/main.rs]: You're completely right, my last commit should fix the wrong behavior. Thanks for checking it out :)

**Pull request merged**